### ISSUE TITLE
Kubernetes agent crashes when non-local APIs are unavailable

### ISSUE THREAD

**Issue by somethingnew2-0** [gravitational/teleport#36252]
Expected behavior:
Kubernetes agent starts up and connects to our Teleport Enterprise cluster. Better error handling in the Teleport Kubernetes agent during the initial Kubernetes API enumeration should not result in a fatal error for unavailable non-local APIs. Ideally, we should still be able to connect to our Kubernetes clusters via Teleport to debug and fix such non-local Kubernetes API errors.

Current behavior:
Kubernetes agent crashes on start with a fatal error because a DataDog non-local metrics API is failing

Bug details:
- Teleport version
We reproduced the issue when upgrading our Teleport Kubernetes agents from 13.4.10 to 14.2.4. The proxy and auth service are already running Enterprise 14.2.4.

- Recreation steps
Have a non-local service failing in your cluster when running a 14.x Teleport Kubernetes agent
```
$ kubectl get --raw /apis/external.metrics.k8s.io/v1beta1
Error from server (ServiceUnavailable): the server is currently unable to handle the request
```
```
$ kubectl get apiservice
NAME                                   SERVICE                                     AVAILABLE                      AGE
...
v1beta1.external.metrics.k8s.io        datadog/datadog-cluster-agent-metrics-api   False (FailedDiscoveryCheck)   305d
...
```

- Debug logs
```
2024-01-02T19:30:36Z ERRO [KUBERNETE] Failed to update cluster schema cluster:cluster_name error:[
ERROR REPORT:
Original Error: *discovery.ErrGroupDiscoveryFailed unable to retrieve the complete list of server APIs: external.metrics.k8s.io/v1beta1: the server is currently unable to handle the request
Stack Trace:
    github.com/gravitational/teleport/lib/kube/proxy/scheme.go:125 github.com/gravitational/teleport/lib/kube/proxy.newClusterSchemaBuilder
    github.com/gravitational/teleport/lib/kube/proxy/cluster_details.go:152 github.com/gravitational/teleport/lib/kube/proxy.newClusterDetails.func1
    runtime/asm_amd64.s:1650 runtime.goexit
User Message: unable to retrieve the complete list of server APIs: external.metrics.k8s.io/v1beta1: the server is currently unable to handle the request] pid:6.1 proxy/cluster_details.go:154

2024-01-02T19:33:50Z WARN [KUBERNETE] Unable to setup context. error:[
ERROR REPORT:
Original Error: *trace.ConnectionProblemError kubernetes cluster &#34;_cluster_name_&#34; is offline
_more stack traces_
```

**Comment by zmb3** [comment]
This was fixed by #36214 and will be out in the next release.

**Comment by somethingnew2-0** [comment]
Oh perfect, thanks!
